src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c

```diff
@@ -20,121 +104,2126 @@ const ActorInit Demo_Effect_InitVars = {
+            this->primXluColor[1] = 0xFF;
+            this->envXluColor[1] = 0xC8;
+            this->envXluColor[0] = 0x00;
+            this->envXluColor[2] = 0x00;
+            this->godLgt.type = God_Lgt_Type_Farore;
+            this->godLgt.rotation = 0;
+            this->initUpdateFunc = &DemoEffect_UpdateGodLgtFarore;
+            this->csActionId = 2;
+            break;
+
+        case Demo_Effect_Light_Ring_Expanding:
+            this->initDrawFunc = &DemoEffect_DrawLightRing;
+            this->initUpdateFunc = &DemoEffect_UpdateLightRingExpanding;
+            this->lightRing.timer = 20;
+            this->lightRing.timerIncrement = 4;
+            this->lightRing.opacity = 255;
+            break;
+
+        case Demo_Effect_Light_Ring_Triforce:
+            this->initDrawFunc = &DemoEffect_DrawLightRing;
+            this->initUpdateFunc = &DemoEffect_UpdateLightRingTriforce;
+            this->lightRing.timer = 20;
+            this->lightRing.timerIncrement = 4;
+            this->lightRing.opacity = 0;
+            this->csActionId = 4;
+            break;
+
+        case Demo_Effect_Light_Ring_Shrinking:
+            this->initDrawFunc = &DemoEffect_DrawLightRing;
+            this->initUpdateFunc = &DemoEffect_UpdateLightRingShrinking;
+            this->lightRing.timer = 351;
+            this->lightRing.timerIncrement = 2;
+            this->lightRing.opacity = 0;
+            break;
+
+        case Demo_Effect_Triforce_Spot:
+            this->initDrawFunc = &DemoEffect_DrawTriforceSpot;
+            this->initUpdateFunc = &DemoEffect_UpdateTriforceSpot;
+            this->triforceSpot.crystalLightOpacity = 0;
+            this->triforceSpot.lightColumnOpacity = 0;
+            this->triforceSpot.triforceSpotOpacity = 0;
+            this->triforceSpot.rotation = 0;
+            this->primXluColor[0] = 0;
+            this->csActionId = 3;
+
+            Actor_SetScale(&this->actor, 0.020f);
+
+            crystalLight = (DemoEffect*)Actor_SpawnAsChild(
+                &globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DEMO_EFFECT, this->actor.posRot.pos.x,
+                this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, Demo_Effect_Crystal_Light);
+
+            if (crystalLight) {
+                Actor_SetScale(&crystalLight->actor, 0.6f);
+            }
+
+            // Necessary to match. Otherwise has function argument reordering.
+            if (globalCtx) {}
+
+            lightRing = (DemoEffect*)Actor_SpawnAsChild(
+                &globalCtx->actorCtx, &crystalLight->actor, globalCtx, ACTOR_DEMO_EFFECT, this->actor.posRot.pos.x,
+                this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, Demo_Effect_Light_Ring_Triforce);
+
+            if (lightRing) {
+                Actor_SetScale(&lightRing->actor, 0.4f);
+            }
+            break;
+
+        case Demo_Effect_Medal_Fire:
+            DemoEffect_InitGetItem(this);
+            this->getItem.giIndex = 12;
```

Comment:
Does it make sense to have an enum for giIndex?

Reply:
Yes, but I think this is out of scope for DemoEffect. It's an index into the gi draw table in z_draw, and that's not fully documented yet.

Reply:
Ah, ok.